app/test/core/services/database/db_service_test.dart

```diff
@@ -59,15 +58,37 @@ void main() {
       when(mockCollectionReference.doc(userModel.userId))
           .thenReturn(mockDocumentReference);
       await dbService.newUser(userModel);
-      verify(mockDocumentReference.set(userModel.toJSON()));
+      verify(mockDocumentReference.set(userModel.toJSON())).called(1);
     });
 
-    test('should add new atr to firestore', () async {
-      final userModel = AnimalTransportRecord.fromJSON(_fakeAtrResponse);
+    test('should set new initial atr to firestore', () async {
+      final userModel = InitialAtr.fromJSON(_fakeInitialAtr, '123');
       when(mockFirestoreInstance.collection(any))
           .thenReturn(mockCollectionReference);
-      await dbService.newRecord(userModel);
-      verify(mockCollectionReference.add(userModel.toJSON()));
+      await dbService.saveNewAtr(userModel);
+      verify(mockCollectionReference.add(userModel.toJSON())).called(1);
+    });
+
+    test('should delete atr to firestore', () async {
+      final userModel = InitialAtr.fromJSON(_fakeInitialAtr, '123');
+      when(mockFirestoreInstance.collection(any))
```

Comment:
Just FYI, you can extract duplicate `when()` statements you need to a helper function. I think it should work the same way (at least it does in Java, Kotlin)